lib/logstash/filters/lookup.rb

```diff
@@ -0,0 +1,183 @@
+# encoding: utf-8
+require 'logstash/filters/base'
+require 'logstash/namespace'
+require 'open-uri'
+require 'digest/sha1'
+require 'json'
+require 'csv'
+
+# A general search and replace tool which uses a Web service with a YAML, CSV or JSON response to determine replacement values.
+#
+# The map entries can be specified with a Web service who your request produces a YML, CSV or JSON response.
+#
+# Operationally, if the event field specified in the `field` configuration
+# matches the EXACT contents of a map entry key, the field's value will be substituted
+# with the matched key's value from the map.
+#
+# By default, the lookup filter will replace the contents of the
+# maching event field (in-place). However, by using the `destination`
+# configuration item, you may also specify a target event field to
+# populate with the new mapd value.
+
+class LogStash::Filters::LookUp < LogStash::Filters::Base
+  config_name "lookup"
+  # The name of the logstash event field containing the value to be compared for a
+  # match by the map filter (e.g. `message`, `host`, `response_code`). 
+  # 
+  # If this field is an array, only the first value will be used.
+  config :field, :validate => :string, :required => true
+
+  # If the destination (or target) field already exists, this configuration item specifies
+  # whether the filter should skip mapping (default) or overwrite the target field
+  # value with the new mapping value.
+  config :override, :validate => :boolean, :default => false
+
+  # The full URI path of a Web service who generates an JSON, yml or CSV format response.
+  # requires to append as suffix the format type. Ex: http://localhost:8080/geoPoints?type=json
+  # http://localhost:8080/geoPoints/json
+  # http://localhost:8080/geoPoints/csv
+  # If no suffix matches, defaults to YAML
+  config :map_url, :validate => :string, :required => true
```

Comment:
renamed to just `url`? `map` is generally a developer facing term and we already have `lookup` as the name of the input. Seems redundant.